wire-ios-data-model/Source/UseCases/HasSelfUserValidE2EICertificatesForAllClientsUseCase.swift

```diff
@@ -0,0 +1,92 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireCoreCrypto
+
+// sourcery: AutoMockable
+/// Determines if the self user has a valid E2EI certificate on all clients.
+public protocol HasSelfUserValidE2EICertificatesForAllClientsUseCaseProtocol {
```

Comment:
What do you think about `IsSelfUserE2EICertifiedUseCaseProtocol`?

Reply:
What about `E2EICertificateForSelfUserUseCaseProtocol` ?

Reply:
But it doesn't return any certificate.

Reply:
Yes, I see. And you created `IsSelfUserVerifiedUseCaseProtocol` for Proteus.
I agree to have it as `IsSelfUserE2EICertifiedUseCaseProtocol`